packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1299,18 +1299,18 @@ class ClipRectLayer extends ContainerLayer {
   /// The [clipRect] argument must not be null before the compositing phase of
   /// the pipeline.
   ///
-  /// The [clipBehavior] argument must not be null, and must not be [Clip.none].
+  /// The [clipBehavior] argument must not be null. If [clipBehavior] is
+  /// [Clip.none], no clipping will be applied.
```

Comment:
Any reason to also apply this change to the layers themselves? Since RenderClipRect and friends no longer push these layers when clipBehavior is none these changes shouldn't really be necessary?

Reply:
I am not sure I follow, can you explain differently?

Reply:
I think you can just revert the changes to this file and everything would still work as you want because the renderobjects don't even push a layer if clipping is set to none.